manager/manager.go

```
@@ -50,6 +51,35 @@ var eventStorageAgeLimit = flag.String("event_storage_age_limit", "default=24h",
 var eventStorageEventLimit = flag.String("event_storage_event_limit", "default=100000", "Max number of events to store (per type). Value is a comma separated list of key values, where the keys are event types (e.g.: creation, oom) or \"default\" and the value is an integer. Default is applied to all non-specified event types")
 var applicationMetricsCountLimit = flag.Int("application_metrics_count_limit", 100, "Max number of application metrics to store (per container)")
 
+// Metrics to be ignored.
+var ignoreMetrics metricsList = metricsList{container.NetworkTcpUsageMetrics: struct{}{}}
+
+func init() {
+	flag.Var(&ignoreMetrics, "disable_metrics", "comma-separated list of metrics to be disabled. Options are `disk`, `network`, `tcp`. Note: tcp is disabled by default due to high CPU usage.")
+}
+
+type metricsList map[container.MetricKind]struct{}
+
+func (ml *metricsList) String() string {
+	return fmt.Sprint(*ml)
+}
+
+func (ml *metricsList) Set(value string) error {
+	if len(*ml) > 0 {
+		return errors.New("disable_metrics already set")
+	}
+	for _, ip := range strings.Split(value, ",") {
+		if ip == container.DiskUsageMetrics.String() {
+			(*ml)[container.DiskUsageMetrics] = struct{}{}
+		} else if ip == container.NetworkUsageMetrics.String() {
+			(*ml)[container.NetworkUsageMetrics] = struct{}{}
+		} else if ip == container.NetworkTcpUsageMetrics.String() {
+			(*ml)[container.NetworkTcpUsageMetrics] = struct{}{}
+		}
```

can we return an error here if the user specified an invalid value ?